packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/ApiKeyAuthImpl.kt

```diff
@@ -0,0 +1,122 @@
+package io.realm.kotlin.mongodb.internal
+
+import io.realm.kotlin.internal.ObjectIdImpl
+import io.realm.kotlin.internal.interop.ObjectIdWrapper
+import io.realm.kotlin.internal.interop.RealmInterop
+import io.realm.kotlin.internal.interop.sync.ApiKeyWrapper
+import io.realm.kotlin.internal.platform.freeze
+import io.realm.kotlin.internal.util.use
+import io.realm.kotlin.mongodb.auth.ApiKey
+import io.realm.kotlin.mongodb.auth.ApiKeyAuth
+import io.realm.kotlin.types.ObjectId
+import kotlinx.coroutines.channels.Channel
+
+internal class ApiKeyAuthImpl(val app: AppImpl, val user: UserImpl) : ApiKeyAuth {
+
+    override suspend fun create(name: String): ApiKey {
+        Channel<Result<ApiKey>>(1).use { channel ->
+            RealmInterop.realm_app_user_apikey_provider_client_create_apikey(
+                app.nativePointer,
+                user.nativePointer,
+                name,
+                channelResultCallback<ApiKeyWrapper, ApiKey>(channel) { apiKeyData ->
+                    ApiKey(
+                        ObjectIdImpl(apiKeyData.id),
+                        apiKeyData.value,
+                        apiKeyData.name,
+                        !apiKeyData.disabled
+                    )
+                }.freeze()
+            )
+            return channel.receive()
+                .getOrThrow()
+        }
+    }
+
+    override suspend fun delete(id: ObjectId) {
```

Comment:
Not sure if that's something you want to do, but in .NET and Dart we ignore the 404 error case for this method as deleting an API key that doesn't exist should be a no-op.

Reply:
Agree that throwing is probably the worst option. We talked about either just ignoring it or returning a boolean indicating whether the deletion was a success. I'm leaning toward returning a boolean as that also mirrors Kotlin/Java collection API's.